src/wallet/rpcdump.cpp

```diff
@@ -694,6 +696,15 @@ UniValue dumpwallet(const JSONRPCRequest& request)
         }
     }
     file << "\n";
+    for (const CScriptID &scriptid : scripts) {
```

Comment:
This is not dumping the information in `m_script_metadata`. At very least, should add a comment with possible todo here to include metadata in the future. Or you could implement a minimal version of this by adding an `EncodeDumpTime(m_script_metadata.at(scriptid)nCreateTime)` field to the output. It would also be possible to extend GetKeyBirthTimes to work with scripts and derive birth times from transaction times, and sort output by these times, to make script output consistent with key output.

Reply:
It would also be good to either dump or add a TODO for dumping WitnessV0ScriptHash and WitnessV0KeyHash keys and scripts.

Reply:
>> It would also be good to either dump or add a TODO for dumping WitnessV0ScriptHash and WitnessV0KeyHash keys and scripts.

> Aren't WitnessV0* scripts and keys also included in mapScripts and mapKeys? Or do you mean like witness derivatives of "normal" keys?

Yeah, they are. I was thinking of mapAddressBook entries, but these aren't exported at all by dumpwallet.